server/serviceClients/authClient/local.js

```diff
@@ -0,0 +1,83 @@
+// This is an example of the simplest scenario when the widget is hosted and used in the same secure domain
+// where as the provider credentials are safe to be stored in the widget server, hence as soon as the auth token is valid
+// Or no auth token needed at all, 
+// simply return the provider credentials, coppied from : 
+// const providerConfig = require('../../../../example/application/configuration')
+
+const {decrypt} = require('../../utils');
+const config = require('../../config');
+
+export class AuthClient {
+  token;
+  constructor(token){
+    this.token = token;
+  }
+
+  async getSecretExchange(iv){
+
+    const key = Buffer.from(config.LocalAuthEncryptionKey, 'base64').toString('hex')
+    const token = decrypt(this.token, key, iv);
+    // if token is valid, or use your own way to validate the token.
+    if(token === config.LocalAuthPhrase){ 
+      return {
+        mxInt: {
+          username: process.env.MxClientId,
```

Comment:
We should probably import these from config.js, so we have a clear place that tells us what we're pulling from .env

Reply:
please read the comments at the top, these are not going into the config because these are not the standard config.  it's not intended to let clients use this provider as is. but we can't provide something that's good enough for anyone, because each clients will have their own standard auth process, whoever whats to use this, have to come to this file, fullly understand it and make their customisations

Reply:
I guess I'm confused about why we are developing this feature if we don't expect anyone to use it.

If we're going to support the use case of not using an auth server and storing credentials within the app server, then it seems like we should make it obvious of how to do that by making example env files and readme descriptions of how to do this. We'll also need end to end tests to make sure it works.

If we're not planning to support this use case, then I don't understand why we're adding it into the code.

Reply:
you need to understand the whole design of the widget , part of the background of this is here https://sophtron.slack.com/archives/C06FK8V1BFW/p1711558034510199
it's said at the top, an example for customers to customize

Reply:
please see readme changes in the original PR, 
https://github.com/Universal-Connect-Project/universal-connect-widget/pull/57/files#diff-679a7f928b73ed9c06f8374dbe256645697332a761d6f6ea7f1b87cfc3a6271d
there is no suitable place for this in this repo, @tphalp mentioned it's being worked on

Reply:
tagged you in the thread,

Reply:
I believe I understand how the 2 different forms of authentication work. I don't understand why we wouldn't want to put some documentation into our readme and example env files to support both forms of authentication

Reply:
I explained that, the proper position in the readme about this is not available and @tphalp is working on it

Reply:
the comments in config were moved to `.env.example`

Reply:
@sophtron-jishun Can you remind me what is still being worked on? I can't remember what we talked about.

Reply:
Yes, what I was referring to is the docs. Our docs team is still working on them. I imagine we will eventually host them on the UCP website.